src/shop-payment-request.html

```diff
@@ -0,0 +1,283 @@
+        },
+
+        summaryLabel: {
+          type: String,
+          value: "Total"
+        },
+
+        _paymentTypesSupported: {
+          computed: '_computePaymentTypesSupported(supportedMethods, androidPay)'
+        }
+      },
+
+      ready: function() {
+        this.listen(window, 'buy-cart', '_onBuyCart');
+        this.listen(window, 'buy-item', '_onBuyItem');
+      },
+
+      _onBuyItem: function(event) {
+        var item = event.detail.item;
+        var itemCost = event.detail.quantity * item.price;
+        var itemDetails = [{
+          label: item.title + ' ' + event.detail.size + ' x' + event.detail.quantity,
+          amount: {currency: this.currency, value: itemCost.toFixed(2)}
+        }];
+
+        this._buyItems(itemDetails, itemCost, false);
+      },
+
+      _onBuyCart: function(event) {
+        if ('PaymentRequest' in window) {
+          var itemDetails = this._computeItemDetails();
+          this._buyItems(itemDetails, this.total, true);
+        } else {
+          this._redirectToCheckout();
+          return;
+        }
+      },
+
+      _buyItems: function(itemDetails, totalCost, cartBuy) {
+        var details = {
+          total: { label: this.summaryLabel, amount:
+            { currency: this.currency, value: totalCost.toFixed(2) }
+          },
+          displayItems: itemDetails
+        };
+
+        // Collect a shipping address, and honor client's other options.
+        var options = {
+          requestShipping: true,
+          requestPayerEmail: this.requestPayerEmail,
+          requestPayerPhone: this.requestPayerPhone
+        };
+
+        try {
+          var request = new PaymentRequest(this._paymentTypesSupported, details, options);
+
+          var that = this;
+
+          // Handle changing shipping options
+          request.addEventListener('shippingoptionchange', function(e) {
+              e.updateWith(new Promise(function(resolve) {
+                  that._updateShippingOptions(details, request.shippingOption, resolve);
+              }));
+          });
+
+          // Handle selecting/changing the shipping address
+          request.addEventListener('shippingaddresschange', function(e) {
+            e.updateWith(new Promise(function(resolve) {
+                that._updateShippingAddress(details, request.shippingAddress, resolve);
+            }));
+          });
+
+          request
+            .show()
+            .then(function(instrumentResponse) {
+              that.fire('process-payment', {
+                'instrument': instrumentResponse,
+                'cartBuy': cartBuy
+              });
+            })
+            .catch(function(err) {
+              if (err === 'Request cancelled') {
+                // Do nothing
+              } else if (err.name === 'NotSupportedError') {
+                that._redirectToCheckout();
+              } else {
+                console.error('Error while handling PaymentRequest.show(): ' + err);
```

Comment:
That error, AFAICT, happens in the case that `window.PaymentRequest` exists but isn't actually implemented (as is the case in Chrome M53+ on !Android if you have `chrome://flags#enable-experimental-web-platform-features` on). A small edge case, for sure, but easy enough to handle. Done.